src/librustc_resolve/late.rs

```diff
@@ -103,7 +92,11 @@ crate enum RibKind<'a> {
     FnItemRibKind,
 
     /// We passed through an item scope. Disallow upvars.
-    ItemRibKind,
+    ItemRibKind {
+        /// Should a "try adding a local generic parameter in this method instead" suggestion be
+        /// added to diagnostics?
+        suggest_adding_parameter: bool
```

Comment:
Nit: can this use a tuple variant like
```rust
enum HasGenericParams { Yes, No }

ItemRibKind(HasGenericParams)
GenericParamsFromOuterFunction(Res, HasGenericParams),
```
I personally hate how ugly and noisy variants with curly braces look.
(Or just `bool` instead of the enum, it's private enough for a comment describing what it means to be good enough.)

Reply:
I've changed it to the suggested enum.

Reply:
`ResolutionError::GenericParamsFromOuterFunction` is not changed.

Reply:
🤦‍♂ Don't know how I managed that, fixed now.